iceberg/src/main/scala/org/apache/spark/sql/delta/serverSidePlanning/IcebergRESTCatalogPlanningClient.scala

```diff
@@ -144,7 +144,7 @@ class IcebergRESTCatalogPlanningClient(
       database: String,
       table: String,
       sparkFilterOption: Option[Filter] = None,
-      projection: Option[StructType] = None): ScanPlan = {
+      sparkProjectionOption: Option[StructType] = None): ScanPlan = {
```

Comment:
should this be full struct type.. or just an array of column names? only the names are needed.. not their types .. right? projection cannot change the column types. 
so if you take the types, you also need to safeguard+test against the types of projected columns being inconsistent with the original types... isnt it?